### DISCUSSION TITLE
Optimization suggestions for returning 301 for requests with different paths for the same target

### DISCUSSION THREAD

**Discussion by hktalent** [projectdiscovery/httpx#719] [General]
@ehsandeep 

Optimization suggestions for returning 301 for requests with different paths for the same target
Optimization suggestions
the same goal, eg
http://target:port/

http://target:port/xx1 301
http://target:port/xx2 301
http://target:port/xx3 301
The now optimization suggests stopping subsequent requests for the same target's premise path

Thanks

**Comment by hktalent** [comment]
The question now is
If the status code of the two pages is 200, which is actually 404, can the similarity be compared with the body of the previous 200 page? If the similarity is greater than or equal to 9.5, both 200s are considered invalid.

Likewise, if the same target is issued before the request
* Can you first request an absolutely non-existent path as absolute 404 template data for subsequent comparisons? If this page returns 301 and 302, do you think the request for the premise page is meaningless?

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
